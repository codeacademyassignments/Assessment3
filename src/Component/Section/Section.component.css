.cardBoundBox{
  display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    border: 6px solid orange;
    border-radius: 47px;
}