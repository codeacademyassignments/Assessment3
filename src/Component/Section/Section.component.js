import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Header from '../Header/Header.component';
import Card from '../Card/Card.component';
import './Section.component.css';

// import ReactDOM from 'react-dom';
// import RawData from '../../RawData/RawData';

class Section extends Component {
  createRows = (blogs) => {
    const rows = [];
    blogs.forEach((blog, index) => {
      rows.push(<Card key={`${index}${blog.Name}`} bookId={blog.id} liked={blog.liked} rating={blog.rating} title={blog.Name} />);
    });
    return rows;
  }

  render() {
    console.log('here......', this.props);
    const { cards, author } = this.props;
    console.log(cards);
    return (
      <section className="cardBoundBox">
        {this.createRows(cards)}
      </section>
    );
  }


  // componentDidUpdate(prevProps) {
  //   console.log('component updated', prevProps);
  // }
}

Section.propTypes = {
  cards: PropTypes.arrayOf(
    PropTypes.shape({
      Name: PropTypes.string,
      id: PropTypes.number,
      rating: PropTypes.float,
      Author: PropTypes.string,
    }),
  ),
  author: PropTypes.string,
};


export default Section;
