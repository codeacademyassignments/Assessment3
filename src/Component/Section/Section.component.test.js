import React from 'react';
import renderer from 'react-test-renderer';
import Section from './Section.component';
import TestData from '../../TestData/TestData';

const blog = TestData['J K Rowling'];
describe('Section', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<Section cards={blog} author="J K Rowling" />);
    expect(tree).toMatchSnapshot();
  });
});
