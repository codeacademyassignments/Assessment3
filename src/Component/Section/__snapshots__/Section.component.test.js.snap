// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshot should match 1`] = `
<section
  className="cardBoundBox"
>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Sorcerers Stone (Harry Potter, #1)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.45
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Chamber of Secrets (Harry Potter, #2)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.38
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.54
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Goblet of Fire (Harry Potter, #4)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.53
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Order of the Phoenix (Harry Potter, #5)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.47
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Half-Blood Prince (Harry Potter, #6)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.54
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
  <article
    className="article"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(cover_image.jpeg)",
        }
      }
    />
    <p
      className="title"
    >
      Harry Potter and the Deathly Hallows (Harry Potter, #7)
    </p>
    <div
      className="cardFooter"
    >
      <div>
        <p
          className="rating"
        >
          4.62
        </p>
      </div>
      <div
        className="heartImage"
      >
        <img
          alt="heart-black"
          className="heart"
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-black.svg",
            }
          }
        />
      </div>
    </div>
  </article>
</section>
`;
